### ISSUE TITLE
Confusing error when using insert as conversion expression

### ISSUE THREAD

**Issue by fmoor** [edgedb/edgedb#4372]
- EdgeDB Version: 2.2 (dgeDB CLI 2.0.5+f78cf34)
- OS Version: ubuntu 22.04

When applying a migration that uses an `insert` as a conversion expression I get an obscure error that seems to complain about the generated migration.

```bash
$ edgedb migration create
did you create link 'org' of object type 'default::Department'? [y,n,l,c,b,s,q,?]
> y
Please specify an expression to populate existing objects in order to make link 'org' of object type 'default::Department' required:
fill_expr> insert Organization { name := "default" }
$ edgedb migration apply
edgedb error: UnsupportedBackendFeatureError: WITH clause containing a data-modifying statement must be at the top level
```

### Schema 1
```
module default {
	type Organization {
		required property name -> str;
	}

	type Department {
		required property name -> str;
	}
}
```

### Schema 2
```
module default {
	type Organization {
		required property name -> str;
	}

	type Department {
		required property name -> str;
		required link org -> Organization;
	}
}
```

**Comment by msullivan** [comment]
This is related to #3592, which suggests that doing this ought to work.
I'll leave this open as an easy "fix the error message" bug, and #3592 can be the issue for really fixing it.

**Comment by msullivan** [comment]
In particular, fixing the error message means catching it in edgedb instead of having it pushed off to postgres and it means catching it while generating migrations instead of just when running them
